roles/pul_nomad/README.md

```diff
@@ -1,36 +1,220 @@
+
+If you need to flush the DNS Cache because IPs have changed, you can restart DNSMasq with `sudo service dnsmasq restart`
+
+## Nomad
+
+Nomad is a container orchestrator.
+
+What this means is you can write job files, tell Nomad to deploy them, and Nomad
+will automatically find a client VM in the cluster that can run them. If that
+client goes down it will launch the container on a different client.
+
+We install Podman on our Nomad client VMs so they can run containers without us
+having to rely on Docker.
+
+### Server Types
+
+Nomad has "server" and "client" installations and configurations.
+
+A cluster consists of 3 server instances, one on each `nomad-host-*.lib.princeton.edu` VM, and a client software install on every client VM (`nomad-client-*.lib.princeton.edu`)
+
+Server installs manage orchestration, receive commands from clients and developers, and
+maintain state about the cluster as a whole. These are named
+`nomad-host-*.princeton.edu`
+
+Client installs run jobs. In our case, containers executed by Podman. These are named
+`nomad-client-*.princeton.edu`
+
+At PUL every Nomad Server VM has a Nomad Server installation and a Consul server installation, and every Nomad Client VM has a Nomad Client installation and a Consul client installation.
+
+### Configuration
+
+Nomad's full configuration reference can be found in their [documentation](https://developer.hashicorp.com/nomad/docs/configuration). You may have to use the table of contents to see details of all the sections.
+
+Every Nomad Client & Server install gets `/etc/nomad.d/base.hcl` created from [`templates/nomad/base.hcl.j2`](templates/nomad/base.hcl.j2).
+
+Every Server install gets `/etc/nomad.d/server.hcl` created from [`templates/nomad/server.hcl.j2`](templates/nomad/server.hcl.j2).
+
+Every Client install gets `/etc/nomad.d/client.hcl` created from [`templates/nomad/client.hcl.j2`](templates/nomad/client.hcl.j2)
+
+The "gossip encryption key" is a key used to encrypt communication between Nomad server installations. We use the same key for both Consul & Nomad.
+
+### Consul Interaction
+
+Nomad uses Consul for clustering (the Nomad server installs find each other by querying
+Consul) and service discovery (when you launch a job in Nomad it can register a
+service in Consul that we can point to from our load balancers.)
+
+### Host Volumes
+
+Host volumes are a way for containers to save data to the underlying client. It's a folder on the client which the container maps - if a job requires a certain host volume it will only ever get scheduled on a node which has that volume. Most containers have no need for persistent data on disk - they store that data in a database or something similar, but if we run a database or something like that in Nomad it might need a host volume. In those cases, we must ensure high availability through some other means (e.g "cloud" mode on Solr, clusters in Postgres, HA Mode in Redis, etc.)
```

Comment:
```suggestion
Host volumes are a way for containers to save data to the underlying client VM. Most containers have no need for persistent data on disk - they store that data in a database or something similar, but if we run a database or something like that in Nomad it might need a host volume. A host volume is a folder on the client which the container maps - if a job (service) requires a certain host volume it will only ever get scheduled on a node which has that volume. In those cases, because we are tying the job (service) to a specific node, we must ensure high availability through some other means (e.g "cloud" mode on Solr, clusters in Postgres, HA Mode in Redis, etc.)
```